host/host.cpp

```diff
@@ -1,33 +1,77 @@
 // Copyright (c) Open Enclave SDK contributors.
 // Licensed under the MIT License.
 
+#include <assert.h>
 #include <iostream>
-#include "../constants/network.h"
-#include "../gen-cpp/RPC.h"
+#include <openenclave/host.h>
 #include <thrift/protocol/TBinaryProtocol.h>
 #include <thrift/server/TSimpleServer.h>
 #include <thrift/transport/TServerSocket.h>
 #include <thrift/transport/TBufferTransports.h>
 
+#include "ortoa_u.h"
+#include "redis.h"
+#include "../constants/constants.h"
+#include "../constants/shared.h"
+#include "../gen-cpp/RPC.h"
+
 using namespace ::apache::thrift;
 using namespace ::apache::thrift::protocol;
 using namespace ::apache::thrift::transport;
 using namespace ::apache::thrift::server;
 
+bool check_simulate(int argc, char *argv[]) {
+  for (int i = 2; i < argc; ++i) {
+    if (strcmp(argv[i], "--simulate") == 0) return true;
+  }
+  return false;
+}
+
 class RPCHandler : virtual public RPCIf {
+ private:
+  inline static uint32_t simulate_flag = OE_ENCLAVE_FLAG_DEBUG;
+  inline static char* oe_enclave_path;
+  inline static oe_enclave_t* enclave;
+  redisCli rd;
+
  public:
   RPCHandler() {
-    // Your initialization goes here
+    oe_result_t result = oe_create_ortoa_enclave(oe_enclave_path, OE_ENCLAVE_TYPE_SGX, simulate_flag, NULL, 0, &enclave);
+    if (result != OE_OK) {
+      std::cerr << "oe_create_ortoa_enclave() failed with enclave path " << oe_enclave_path << " " << result << std::endl;
+      throw "oe ortoa enclave init failed";
```

Comment:
Can we throw an actual error type

Reply:
Yep, will be dealing with error handling as a whole once merged into client.